src/views/v2/commerce/checkout/cart/billing-details.php

```diff
@@ -12,33 +12,31 @@
  * @since   TBD
  *
  * @version TBD
- * @var bool $must_login [Global] Whether login is required to buy tickets or not.
+ * @var bool     $must_login [Global] Whether login is required to buy tickets or not.
+ * @var \WP_User $user       The current user logged in, or a non-user w/ ID 0.
  */
 
 if ( $must_login ) {
 	return;
 }
 
-$user      = wp_get_current_user();
-$username  = '';
-$useremail = '';
-
-if ( ! empty( $user ) ) {
-	$username  = $user->display_name;
-	$useremail = $user->user_email;
-}
+$username  = ! empty( $user->display_name ) ? $user->display_name : '';
+$useremail = ! empty( $user->user_email ) ? $user->user_email : '';
 
 ?>
 
 <div id="tec-tc-gateway-stripe-billing-identification">
 	<h4><?php esc_html_e( 'Billing Information', 'event-tickets' ); ?></h4>
 	<span id="tec-tc-gateway-stripe-billing-first-name">
-		<input type="text" value="<?php echo esc_attr( $username ); ?>" placeholder="First Name"/>
+		<input type="text" value="<?php echo esc_attr( $username ); ?>"
+			   placeholder="<?php esc_html_e( 'First Name', 'event-tickets' ); ?>"/>
```

Comment:
I think this should be `esc_attr_e`. We're using the same var for first and last name.